py/desispec/scripts/procexp.py

```diff
@@ -92,17 +92,16 @@ def main(args):
             # copy mask
             frame.mask = copied_frame.mask
             
-            if args.sky_throughput_correction :
+            if not args.no_sky_throughput_correction :
                 # and (re-)subtract sky, but just the correction term
                 subtract_sky(frame, skymodel, throughput_correction = True)
```

Comment:
This predates this PR, but I'm confused about the logic here.  If cosmics_nsig>0 and args.no_sky_throughput_correction is True, then doesn't this mean that the skymodel is never subtracted from the original frame?  I don't understand the comment "and (re-)subtract sky, but just the correction term".

Reply:
I think you're right.

Line 80 has
```
           # first subtract sky without throughput correction                                                                                             
            subtract_sky(copied_frame, skymodel, apply_throughput_correction = False)
```

But that is only on the copied frame, which only affects the mask. The actual frame is only handed to `subtract_sky` *if* `sky_throughput_correction` is `True`  (`no_sky_throughput_correction` is `False`).

I think we can turn lines 95-97  to a single line identical to line 100:
`subtract_sky(frame, skymodel, apply_throughput_correction = (not args.no_sky_throughput_correction) )`

@julienguy -- comments?

Reply:
It's indeed a bug I introduced recently when fixing another one.

Reply:
I changed this to what I mentioned in the comment above. removing the if statement and directly setting apply_throughput_correction using the flag